config/config.go

```diff
@@ -193,104 +193,104 @@ func MustNewRegexp(s string) Regexp {
+	PreferredIPProtocol string           `yaml:"preferred_ip_protocol,omitempty" json:"preferred_ip_protocol,omitempty"`
 }
 
 type HeaderMatch struct {
-	Header       string `yaml:"header,omitempty"`
-	Regexp       Regexp `yaml:"regexp,omitempty"`
-	AllowMissing bool   `yaml:"allow_missing,omitempty"`
+	Header       string `yaml:"header,omitempty" json:"header,omitempty"`
+	Regexp       Regexp `yaml:"regexp,omitempty" json:"regexp,omitempty"`
+	AllowMissing bool   `yaml:"allow_missing,omitempty" json:"allow_missing,omitempty"`
 }
 
 type Label struct {
-	Name  string `yaml:"name,omitempty"`
-	Value string `yaml:"value,omitempty"`
+	Name  string `yaml:"name,omitempty" json:"name,omitempty"`
+	Value string `yaml:"value,omitempty" json:"value,omitempty"`
 }
 
 type QueryResponse struct {
-	Expect   Regexp  `yaml:"expect,omitempty"`
-	Labels   []Label `yaml:"labels,omitempty"`
-	Send     string  `yaml:"send,omitempty"`
-	StartTLS bool    `yaml:"starttls,omitempty"`
+	Expect   Regexp  `yaml:"expect,omitempty" json:"expect,omitempty"`
+	Labels   []Label `yaml:"labels,omitempty" json:"labels,omitempty"`
+	Send     string  `yaml:"send,omitempty" json:"send,omitempty"`
+	StartTLS bool    `yaml:"starttls,omitempty" json:"starttls,omitempty"`
 }
 
 type TCPProbe struct {
-	IPProtocol         string           `yaml:"preferred_ip_protocol,omitempty"`
-	IPProtocolFallback bool             `yaml:"ip_protocol_fallback,omitempty"`
-	SourceIPAddress    string           `yaml:"source_ip_address,omitempty"`
-	QueryResponse      []QueryResponse  `yaml:"query_response,omitempty"`
-	TLS                bool             `yaml:"tls,omitempty"`
-	TLSConfig          config.TLSConfig `yaml:"tls_config,omitempty"`
+	IPProtocol         string           `yaml:"preferred_ip_protocol,omitempty" json:"preferred_ip_protocol,omitempty"`
+	IPProtocolFallback bool             `yaml:"ip_protocol_fallback,omitempty" json:"ip_protocol_fallback,omitempty"`
+	SourceIPAddress    string           `yaml:"source_ip_address,omitempty" json:"source_ip_address,omitempty"`
+	QueryResponse      []QueryResponse  `yaml:"query_response,omitempty" json:"query_response,omitempty"`
+	TLS                bool             `yaml:"tls,omitempty" json:"tls,omitempty"`
+	TLSConfig          config.TLSConfig `yaml:"tls_config,omitempty" json:"tls_config,omitempty"`
 }
 
 type ICMPProbe struct {
-	IPProtocol         string `yaml:"preferred_ip_protocol,omitempty"` // Defaults to "ip6".
-	IPProtocolFallback bool   `yaml:"ip_protocol_fallback,omitempty"`
-	SourceIPAddress    string `yaml:"source_ip_address,omitempty"`
-	PayloadSize        int    `yaml:"payload_size,omitempty"`
-	DontFragment       bool   `yaml:"dont_fragment,omitempty"`
-	TTL                int    `yaml:"ttl,omitempty"`
+	IPProtocol         string `yaml:"preferred_ip_protocol,omitempty"` // Defaults to "ip6" json:"preferred_ip_protocol,omitempty"` // Defaults to "ip6".
```

Comment:
comment is now in between tags, happens at few other places as well.

```suggestion
	IPProtocol         string `yaml:"preferred_ip_protocol,omitempty" json:"preferred_ip_protocol,omitempty"` // Defaults to "ip6".
```

Reply:
Fixed and adjusted the SED I used.